optexp/datasets/tokenizers.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+from tqdm import tqdm
+
+from optexp.component import Component
+import sentencepiece as sp
+import torch
+
+
+class Tokenizer(ABC, Component):
+
+    @abstractmethod
+    def build_tokenizer(self, model_path: Path, data_path: Path, vocab_size: int):
```

Comment:
`model_path` => `tokenizer_path` to avoid confusion with the model?

Reply:
done